src/test/suite/vscode-integration-specific.test.ts

```diff
@@ -531,4 +531,25 @@ describe("VSCode integration specific stuff", () => {
         await wait(200);
         await assertContent({ cursor: [2, 2] }, client);
     });
+
+    it("vim.ui.select works", async () => {
+        await openTextDocument({ content: "" });
+
+        const p1 = client.lua(`
+            vim.ui.select({ 'red', 'green', 'blue' }, {}, function(item, idx)
+                local buffer = vim.api.nvim_get_current_buf()
+                vim.api.nvim_buf_set_lines(buffer, 0, -1, false, {item.label})
```

Comment:
For this type of test, you can store the result in a temporary variable, and then assert it.

nvim: vim.g._ret = ...
vscode: client.getVar ...